packages/near-membrane-base/src/membrane.ts

```diff
@@ -754,17 +852,57 @@ export function createMembraneMarshall() {
                 this.preventExtensions = BoundaryProxyHandler.defaultPreventExtensionsTrap;
                 this.setPrototypeOf = BoundaryProxyHandler.defaultSetPrototypeOfTrap;
                 this.set = BoundaryProxyHandler.defaultSetTrap;
-                if (foreignTargetTraits & TargetTraits.Revoked) {
+                if (isRevoked) {
                     revoke();
+                } else if (isUnrevokedObject) {
+                    // Lazily define unbrandedTag.
+                    let unbrandedTag: string | undefined | symbol =
+                        LOCKER_NEAR_MEMBRANE_UNDEFINED_VALUE_SYMBOL;
+                    ReflectApply(ObjectProto__defineGetter__, this, [
+                        'unbrandedTag',
+                        () => {
```

Comment:
same

Reply:
The unbranded tag is to help those objects like regexp, date, typed arrays, and others that when proxy wrapped and passed to `Object.prototype.toString.call` will produce `[object Object]` instead of things like `[object RegExp]`. Since `[object Object]` is the behavior of proxies anyways we don't project for them if that's the resolve value (there's no need to expose magic in that case). We only project if the value doesn't have it's own `Symbol.toStringTag` value and is something other than `[object Object]`. This is a more narrow case and something that isn't required for all-the-things up-front. That said, we could also not memoize this check.